ci/ci/build.py

```diff
@@ -876,38 +884,65 @@ def build(self, batch, code, scope):  # pylint: disable=unused-argument
             '_name': self._name,
             'admin_username': self.admin_username,
             'user_username': self.user_username,
+            'admin_password_file': self.admin_password_file,
+            'user_password_file': self.user_password_file,
             'cant_create_database': self.cant_create_database,
             'migrations': self.migrations,
             'shutdowns': self.shutdowns
         }
 
-        create_script = f'''
+        create_password_script = f'''
+set -ex
+
+LC_ALL=C tr -dc '[:alnum:]' </dev/urandom | head -c 16 > {self.admin_password_file}
+LC_ALL=C tr -dc '[:alnum:]' </dev/urandom | head -c 16 > {self.user_password_file}
+'''
+
+        create_database_script = f'''
 set -ex
 
 python3 create_database.py {shq(json.dumps(create_database_config))}
 '''
 
+        input_files = []
         if self.inputs:
-            input_files = []
             for i in self.inputs:
                 input_files.append((f'{BUCKET}/build/{batch.attributes["token"]}{i["from"]}', i["to"]))
-        else:
-            input_files = None
-
-        self.job = batch.create_job(CI_UTILS_IMAGE,
-                                    command=['bash', '-c', create_script],
-                                    attributes={'name': self.name},
-                                    secrets=[{
-                                        'namespace': self.database_server_config_namespace,
-                                        'name': 'database-server-config',
-                                        'mount_path': '/sql-config'
-                                    }],
-                                    service_account={
-                                        'namespace': BATCH_PODS_NAMESPACE,
-                                        'name': 'ci-agent'
-                                    },
-                                    input_files=input_files,
-                                    parents=self.deps_parents())
+        password_files_input = [
+            (f'{BUCKET}/build/{batch.attributes["token"]}/{self.admin_password_file}', self.admin_password_file),
+            (f'{BUCKET}/build/{batch.attributes["token"]}/{self.user_password_file}', self.user_password_file)]
+        input_files.extend(password_files_input)
+
+        self.create_password_job = batch.create_job(
+            CI_UTILS_IMAGE,
+            command=['bash', '-c', create_password_script],
+            attributes={'name': self.name},
+            secrets=[{
+                'namespace': self.database_server_config_namespace,
+                'name': 'database-server-config',
+                'mount_path': '/sql-config'
+            }],
+            service_account={
+                'namespace': BATCH_PODS_NAMESPACE,
+                'name': 'ci-agent'
+            },
+            output_files=[(x[1], x[0]) for x in password_files_input])
```

Comment:
Should the parents of this job just be the deps_parents? Then the create_database job can just depend on this job.

Reply:
sure